components/event-bus/internal/knative/util/kn-eventing.go

```diff
@@ -176,7 +179,7 @@ func (k *KnativeLib) CreateChannel(name string, namespace string, labels *map[st
 		case <-tout:
 			return nil, errors.New("timed out")
 		case <-tick:
-			if channel, err = k.messagingChannel.Channels(namespace).Get(name, metav1.GetOptions{}); err != nil {
+			if channel, err = k.GetChannelByLabels(namespace, &labels); err != nil {
```

Comment:
Another pointer.

Reply:
GetChannelByLabels() is still taking a pointer. i dont' intend to change this for this PR, the other PR can take care of this, right?

Reply:
Sure but the API is now inconsistent.

Reply:
not really, GetChannelByLabels() is consistent throughout the source, as the interface definition itself takes pointer. so it can be fixed once for all in that other PR

Reply:
I mean that the same data structure is passed to the same interface methods in multiple different manners (sometimes value, sometimes reference) without any consistency.

I'm not in favor in partial fixes that leave half of the work to other efforts. Why not just avoiding the change then?

Reply:
done